contracts/mainnet/connectors/pooltogether/main.sol

```diff
@@ -0,0 +1,203 @@
+pragma solidity ^0.7.0;
+
+/**
+ * @title PoolTogether
+ * @dev Deposit & Withdraw from PoolTogether
+ */
+
+ import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+ import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+ import { PrizePoolInterface, TokenFaucetInterface, TokenFaucetProxyFactoryInterface, PodInterface } from "./interface.sol";
+
+import { TokenInterface } from "../../common/interfaces.sol";
+import { Events } from "./events.sol";
+import { DSMath } from "../../common/math.sol";
+import { Basic } from "../../common/basic.sol";
+
+abstract contract PoolTogetherResolver is Events, DSMath, Basic {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @dev Deposit into Prize Pool
+     * @notice Deposit assets into the Prize Pool in exchange for tokens
+     * @param prizePool PrizePool address to deposit to
+     * @param to The address receiving the newly minted tokens
+     * @param amount The amount of the underlying asset the user wishes to deposit. The Prize Pool contract should have been pre-approved by the caller to transfer the underlying ERC20 tokens.
+     * @param controlledToken The address of the token that they wish to mint. For our default Prize Strategy this will either be the Ticket address or the Sponsorship address.  Those addresses can be looked up on the Prize Strategy.
+     * @param referrer The address that should receive referral awards, if any.
+     * @param getId Get token amount at this ID from `InstaMemory` Contract.
+     * @param setId Set token amount at this ID in `InstaMemory` Contract.
+    */
+
+    function depositTo(
+        address prizePool,
+        address to,
```

Comment:
@eccheung4 Tokens will always be minted to DSA wallet only. So, we can replace `to` with `address(this)`.